compiler/infra/ForwardList.hpp

```diff
@@ -0,0 +1,684 @@
+#ifndef FORWARD_LIST_HPP
+#define FORWARD_LIST_HPP
+
+#pragma once
+
+#if 0
```

Comment:
Right. This means to never hold a pointer to the base class. I don't know how to enforce that.
I am also thinking that if STL containers were never meant to be extended, future implementations may add the keyword 'final' to the class making the extension impossible.